File: experiments/dbx-tools-u2m/crates/dbx-tools-u2m-python/examples/token.py
import os

import asyncio

from dbx_tools_u2m_bindings import U2mOptions, create_persistent_auth


async def main() -> None:
    auth = await create_persistent_auth(
        U2mOptions(profile=os.getenv("DATABRICKS_CONFIG_PROFILE"))
    )
    token = await auth.token()
    status = auth.status()
    print(
        {
            "profile": status.profile,
            "host": status.host,
            "token_type": token.token_type,
            "expiry": token.expiry,
        }
    )


asyncio.run(main())

File: experiments/dbx-tools-u2m/crates/dbx-tools-u2m-node/examples/token.mjs
import { createPersistentAuth, U2mOptions } from "../dist/index.js";

const auth = await createPersistentAuth(
  U2mOptions.create({ profile: process.env.DATABRICKS_CONFIG_PROFILE }),
);
const token = await auth.token();
const status = auth.status();
console.log({
  profile: status.profile,
  host: status.host,
  tokenType: token.tokenType,
  expiry: token.expiry,
});
